File: archive/cdk-py/Pr2381/lib/cdk/vpc_peering_stack.py
from aws_cdk import aws_ec2 as ec2, Stack
from constructs import Construct


class VpcPeeringStack(Stack):
  def __init__(
    self,
    scope: Construct,
    stack_id: str,
    vpc1: ec2.Vpc,
    vpc2: ec2.Vpc,
    **kwargs
  ):
    super().__init__(scope, stack_id, **kwargs)

    peering = ec2.CfnVPCPeeringConnection(
      self,
      "Peering",
      vpc_id=vpc1.vpc_id,
      peer_vpc_id=vpc2.vpc_id,
      peer_region="us-east-2",
    )

    # Add routes for VPC1 private subnets
    for index, subnet in enumerate(vpc1.private_subnets):
      ec2.CfnRoute(
        self,
        f"RouteToVPC2-Subnet{index}",
        route_table_id=subnet.route_table.route_table_id,
        destination_cidr_block=vpc2.vpc_cidr_block,
        vpc_peering_connection_id=peering.ref,
      )

    # Add routes for VPC2 private subnets
    for index, subnet in enumerate(vpc2.private_subnets):
      ec2.CfnRoute(
        self,
        f"RouteToVPC1-Subnet{index}",
        route_table_id=subnet.route_table.route_table_id,
        destination_cidr_block=vpc1.vpc_cidr_block,
        vpc_peering_connection_id=peering.ref,
      )

File: archive/cdk-ts/Pr5857/lib/vpc-peering-stack.ts
import * as cdk from 'aws-cdk-lib';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

export interface VpcPeeringStackProps extends cdk.StackProps {
  environmentSuffix: string;
  sourceVpc: ec2.IVpc;
  targetVpc: ec2.IVpc;
  sourceVpcCidr: string;
  targetVpcCidr: string;
}

export class VpcPeeringStack extends cdk.Stack {
  public readonly peeringConnection: ec2.CfnVPCPeeringConnection;

  constructor(scope: Construct, id: string, props: VpcPeeringStackProps) {
    super(scope, id, props);

    const {
      environmentSuffix,
      sourceVpc,
      targetVpc,
      sourceVpcCidr,
      targetVpcCidr,
    } = props;

    // Create VPC Peering Connection
    this.peeringConnection = new ec2.CfnVPCPeeringConnection(
      this,
      `PeeringConnection-${environmentSuffix}`,
      {
        vpcId: sourceVpc.vpcId,
        peerVpcId: targetVpc.vpcId,
        tags: [
          {
            key: 'Name',
            value: `vpc-peering-${environmentSuffix}`,
          },
          {
            key: 'EnvironmentSuffix',
            value: environmentSuffix,
          },
        ],
      }
    );

    // Add routes in source VPC to target VPC
    sourceVpc.privateSubnets.forEach((subnet, index) => {
      new ec2.CfnRoute(this, `SourceRoute${index}-${environmentSuffix}`, {
        routeTableId: subnet.routeTable.routeTableId,
        destinationCidrBlock: targetVpcCidr,
        vpcPeeringConnectionId: this.peeringConnection.ref,
      });
    });

    // Add routes in target VPC to source VPC
    targetVpc.privateSubnets.forEach((subnet, index) => {
      new ec2.CfnRoute(this, `TargetRoute${index}-${environmentSuffix}`, {
        routeTableId: subnet.routeTable.routeTableId,
        destinationCidrBlock: sourceVpcCidr,
        vpcPeeringConnectionId: this.peeringConnection.ref,
      });
    });

    new cdk.CfnOutput(this, 'PeeringConnectionId', {
      value: this.peeringConnection.ref,
      description: 'VPC Peering Connection ID',
      exportName: `VpcPeeringId-${environmentSuffix}`,
    });
  }
}
